src/nvim/api/vim.c

```diff
@@ -2389,3 +2389,151 @@ Dictionary nvim_complete_set(Integer index, Dict(complete_set) *opts, Arena *are
+    *flush = true;
+  }
+
+  if (*flush && (status & (kWinBar|kStatusLine))) {
+    // When flushing later, just mark for redraw.
+    wp->w_redr_status = true;
+  } else if (status & (kWinBar|kStatusLine)) {
+    // Not flushing later, only redraw the 'winbar' or 'statusline'.
+    win_check_ns_hl(wp);
+    if (status & kWinBar) {
+      win_redr_winbar(wp);
+    }
+    if (status & kStatusLine) {
+      win_redr_status(wp);
+    }
+    win_check_ns_hl(NULL);
+  }
+}
+
+/// Instruct Nvim to redraw various components.
+///
+/// @param opts  Optional parameters.
+///               - win: Target a specific |window-ID| for various components.
+///               - buf: Target a specific buffer number for various
+///                 components.
+///               - flush: Update the screen with pending updates, even if
+///                 Nvim is waiting for input and would normally not do so.
+///               - clear: Forcefully redraw `win`, all windows containing
+///                 `buf`, or all windows.
+///               - range: Buffer range to redraw, useful for decoration
+///                 providers. Expects a tuple `[first, last]` with the first
+///                 and last line number of the range. Targets `buf`, the
+///                 buffer in `win` or the current buffer.
+///               - cursor: Immediately update cursor position on the screen
+///                 in window `win` or the current window.
+///               - statuscolumn: Redraw the |'statuscolumn'| in `buf`, `win`
+///                 or all windows.
+///               - statusline: Redraw the |'statusline'| in `buf`, `win` or
+///                 all windows.
+///               - winbar: Redraw the |'winbar'| in `buf`, `win` or all
+///                 windows.
+///               - tabline: Redraw the |'tabline'|.
+void nvim_redraw(Dict(redraw) *opts, Error *err)
+  FUNC_API_SINCE(12)
+{
+  win_T *win = NULL;
+  buf_T *buf = NULL;
+  bool save_lz = p_lz;
+  int save_rd = RedrawingDisabled;
+  RedrawingDisabled = 0;
+  p_lz = false;
+
+  if (HAS_KEY(opts, redraw, win)) {
+    win = find_window_by_handle(opts->win, err);
+    if (ERROR_SET(err)) {
+      goto theend;
+    }
+  }
+
+  if (HAS_KEY(opts, redraw, buf)) {
+    VALIDATE(win == NULL, "%s", "Invalid 'buf': already specified 'win'.", {
+      goto theend;
+    });
+    buf = find_buffer_by_handle(opts->buf, err);
+    if (ERROR_SET(err)) {
+      goto theend;
+    }
+  }
+
+  if (opts->clear) {
+    if (win != NULL) {
+      redraw_later(win, UPD_NOT_VALID);
+    } else if (buf != NULL) {
+      redraw_buf_later(buf, UPD_NOT_VALID);
+    } else {
+      redraw_all_later(UPD_NOT_VALID);
+    }
+  }
+
+  if (HAS_KEY(opts, redraw, range)) {
+    VALIDATE(kv_size(opts->range) == 2
+             && kv_A(opts->range, 0).type == kObjectTypeInteger
+             && kv_A(opts->range, 1).type == kObjectTypeInteger,
+             "%s", "Invalid 'range': expected 2-tuple of Lua numbers", {
```

Comment:
```suggestion
             "%s", "Invalid 'range': expected 2-tuple of Integers", {
```

Reply:
Seemed more logical to me as well but then this should be updated first I think:
https://github.com/neovim/neovim/blob/8697f3274b240a5a00774dba90329c00c4e914ab/src/nvim/lua/converter.c#L829